From e38b6114fb02f14ed51870e68116d336aa462add Mon Sep 17 00:00:00 2001
From: gaulthier gain <gaulthier.gain@uliege.be>
Date: Sun, 18 Oct 2020 10:26:45 +0000
Subject: [PATCH] Fix stdout flushing in musl

In this patch, we temporary remove the call to `ioctl` in
`__stdout_write.c` which causes an assertion error due to invalid
argument order. In addition, `__stdio_write` is fixed by updating
a condition to avoid an infinite loop since the last caracter of the
buffer is skipped when the `cnt` variable is tested with the
`iov_len` variable.

Signed-off-by: gaulthier gain <gaulthier.gain@uliege.be>
---
 src/stdio/__stdio_write.c  | 2 +-
 src/stdio/__stdout_write.c | 2 --
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/stdio/__stdio_write.c b/src/stdio/__stdio_write.c
index d2d8947..6ebf59e 100644
--- a/src/stdio/__stdio_write.c
+++ b/src/stdio/__stdio_write.c
@@ -24,7 +24,7 @@ size_t __stdio_write(FILE *f, const unsigned char *buf, size_t len)
 			return iovcnt == 2 ? 0 : len-iov[0].iov_len;
 		}
 		rem -= cnt;
-		if (cnt > iov[0].iov_len) {
+		if (cnt >= iov[0].iov_len) {
 			cnt -= iov[0].iov_len;
 			iov++; iovcnt--;
 		}
diff --git a/src/stdio/__stdout_write.c b/src/stdio/__stdout_write.c
index dd1ec60..1fdc12e 100644
--- a/src/stdio/__stdout_write.c
+++ b/src/stdio/__stdout_write.c
@@ -5,7 +5,5 @@ size_t __stdout_write(FILE *f, const unsigned char *buf, size_t len)
 {
 	struct winsize wsz;
 	f->write = __stdio_write;
-	if (!(f->flags & F_SVB) && __syscall(SYS_ioctl, f->fd, TIOCGWINSZ, &wsz))
-		f->lbf = -1;
 	return __stdio_write(f, buf, len);
 }
-- 
2.17.1

